chalicelib/checks/wrangler_checks.py

```diff
@@ -1842,3 +1842,30 @@ def add_suggested_enum_values(connection, **kwargs):
     action.status = 'DONE'
     action.output = action_logs
     return action
+
+
+@check_function()
+def check_external_references_uri(connection, **kwargs):
+    '''
+    Check if external_references.uri is missing while external_references.ref
+    is present. Note that this check does not detect cases where multiple
+    references are present but not all have problems.
+    '''
```

Comment:
If we had nested mapping in fourfront it would be easy peasy to do this with a simple search.  I still think it might be worthwhile to try to check all cases, which I think we can do by only looking at the external_references property.  Something like:
search = "/search/?type=Item&external_references.ref%21=No+value&field=external_references"
result = ff_utils.search_metadata(search, key=connection.ff_keys)
for res in result:
    baddies = [er for er in res.get('external_references', []) if not er.get('uri')]
    if baddies:
             #report them

Parameters to search_metadata - eg. make it a generator with a page limit if needed.